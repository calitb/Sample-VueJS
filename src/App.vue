<template>
  <div>
    <router-view />
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
import getItems from '@/api/getItems';

export default Vue.extend({
  name: 'App',
  methods: {
    loadItems() {
      getItems((error, data) => {
        if (!error) {
          this.$root.$data.items = data;
        }
      });
    }
  },
  created() {
    this.loadItems();
  }
});
</script>
