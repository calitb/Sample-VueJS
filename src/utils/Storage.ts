export function removeItem(key: string) {
  sessionStorage.removeItem(key);
}

export function getItem(key: string): string | null {
  return sessionStorage.getItem(key);
}

export function setItem(key: string, value: string) {
  sessionStorage.setItem(key, value);
}
