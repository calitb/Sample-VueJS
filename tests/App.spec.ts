import { mount } from "@vue/test-utils";
import App from "@/App.vue";

import Vuex from "vuex";
import { createLocalVue } from "./utils";

describe("App", () => {
  describe("Default", () => {
    const fetchItemsAction = jest.fn();
    const localVue = createLocalVue();
    const store = new Vuex.Store({
      actions: {
        ["FETCH_ITEMS_ACTION"]: fetchItemsAction
      }
    });

    const wrapper = mount(App, {
      localVue,
      store,
      stubs: {
        "router-view": true
      }
    });

    it("should render the component", () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it("should call fetchItems on load", () => {
      expect(fetchItemsAction).toBeCalled();
    });
  });
});
