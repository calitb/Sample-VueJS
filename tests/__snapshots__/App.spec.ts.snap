// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Default should render the component 1`] = `
<div>
  <router-view-stub />
</div>
`;

exports[`App No items because of fetch error should render the component 1`] = `
<div>
  <router-view-stub />
</div>
`;
